internal/manifest/manifest.go

```diff
@@ -140,40 +147,94 @@ func (m *Manifest) Validate() error {
+		if err := check(report); err != nil {
+			return nil, fmt.Errorf("additional check failed: %w", err)
+		}
+	}
+	return g.opts, nil
+}
+
+// TODO(msanft): add generic validation interface for other attestation types.
+
+// SNPValidateOpts returns validate options generators populated with the manifest's
+// SNP reference values and trusted measurement for the given runtime.
+func (m *Manifest) SNPValidateOpts() ([]*SNPValidateOptsGenerator, error) {
+	if m.ReferenceValues == nil {
+		return nil, errors.New("reference values cannot be empty")
 	}
 
 	if err := m.Validate(); err != nil {
 		return nil, fmt.Errorf("validating manifest: %w", err)
 	}
-	trustedMeasurement, err := m.ReferenceValues.AKS.TrustedMeasurement.Bytes()
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert TrustedMeasurement from manifest to byte slices: %w", err)
-	}
-
-	return &validate.Options{
-		Measurement: trustedMeasurement,
-		GuestPolicy: abi.SnpPolicy{
-			Debug: false,
-			SMT:   true,
-		},
-		VMPL: new(int), // VMPL0
-		MinimumTCB: kds.TCBParts{
-			BlSpl:    m.ReferenceValues.AKS.SNP.MinimumTCB.BootloaderVersion.UInt8(),
-			TeeSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.TEEVersion.UInt8(),
-			SnpSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.SNPVersion.UInt8(),
-			UcodeSpl: m.ReferenceValues.AKS.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
-		},
-		MinimumLaunchTCB: kds.TCBParts{
-			BlSpl:    m.ReferenceValues.AKS.SNP.MinimumTCB.BootloaderVersion.UInt8(),
-			TeeSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.TEEVersion.UInt8(),
-			SnpSpl:   m.ReferenceValues.AKS.SNP.MinimumTCB.SNPVersion.UInt8(),
-			UcodeSpl: m.ReferenceValues.AKS.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
-		},
-		PermitProvisionalFirmware: true,
-	}, nil
+
+	var out []*SNPValidateOptsGenerator
+	for _, refVal := range m.ReferenceValues {
+		trustedMeasurement, err := refVal.SNP.TrustedMeasurement.Bytes()
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert TrustedMeasurement from manifest to byte slices: %w", err)
+		}
+
+		out = append(out, &SNPValidateOptsGenerator{
+			manifest: m,
+			opts: &validate.Options{
+				Measurement: trustedMeasurement,
+				GuestPolicy: abi.SnpPolicy{
+					Debug: false,
+					SMT:   true,
+				},
+				VMPL: new(int), // VMPL0
+				MinimumTCB: kds.TCBParts{
+					BlSpl:    refVal.SNP.MinimumTCB.BootloaderVersion.UInt8(),
+					TeeSpl:   refVal.SNP.MinimumTCB.TEEVersion.UInt8(),
+					SnpSpl:   refVal.SNP.MinimumTCB.SNPVersion.UInt8(),
+					UcodeSpl: refVal.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
+				},
+				MinimumLaunchTCB: kds.TCBParts{
+					BlSpl:    refVal.SNP.MinimumTCB.BootloaderVersion.UInt8(),
+					TeeSpl:   refVal.SNP.MinimumTCB.TEEVersion.UInt8(),
+					SnpSpl:   refVal.SNP.MinimumTCB.SNPVersion.UInt8(),
+					UcodeSpl: refVal.SNP.MinimumTCB.MicrocodeVersion.UInt8(),
+				},
+				PermitProvisionalFirmware: true,
+			},
+		})
+	}
+
+	if len(out) == 0 {
+		return nil, errors.New("no AKS reference values found in manifest")
```

Comment:
Is this a problem? If a user only wants to use TDX, this seems correct.

Reply:
If a user only wants to use TDX, we won't call `SNPValidateOpts`, I think.

Reply:
AFAICT that's not currently the case.

Reply:
We don't even have TDX validation implemented yet. You're unable to grab `main` and spawn a validatable TDX cluster from it now, and I don't think we should use this PR to pave the way to be able to do so.